provider/dynamodb_test.go

```diff
@@ -0,0 +1,105 @@
+package provider
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbattribute"
+	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbiface"
+	"github.com/containous/traefik/types"
+	"reflect"
+	"testing"
+)
+
+type mockDynamoDBCLient struct {
+	dynamodbiface.DynamoDBAPI
+}
+
+var backend = &types.Backend{
+	HealthCheck: &types.HealthCheck{
+		URL: "/build",
+	},
+	Servers: map[string]types.Server{
+		"server1": types.Server{
+			URL: "http://test.traefik.io",
+		},
+	},
+}
+
+var frontend = &types.Frontend{
+	EntryPoints: []string{"http"},
+	Backend:     "test.traefik.io",
+	Routes: map[string]types.Route{
+		"route1": types.Route{
+			Rule: "Host:test.traefik.io",
+		},
+	},
+}
+
+// ScanPages simulates a call to ScanPages (see https://docs.aws.amazon.com/sdk-for-go/api/service/dynamodb/#DynamoDB.ScanPages)
+// by running the fn function twice and returning an item each time.
+func (m *mockDynamoDBCLient) ScanPages(input *dynamodb.ScanInput, fn func(*dynamodb.ScanOutput, bool) bool) error {
+	attributeBackend, err := dynamodbattribute.Marshal(backend)
+	if err != nil {
+		return err
+	}
+
+	attributeFrontend, err := dynamodbattribute.Marshal(frontend)
+	if err != nil {
+		return err
+	}
+
+	fn(&dynamodb.ScanOutput{
+		Items: []map[string]*dynamodb.AttributeValue{
+			{
+				"id": &dynamodb.AttributeValue{
+					S: aws.String("test.traefik.io_backend"),
+				},
+				"name": &dynamodb.AttributeValue{
+					S: aws.String("test.traefik.io"),
+				},
+				"backend": attributeBackend,
+			},
+		},
+	}, false)
+
+	fn(&dynamodb.ScanOutput{
+		Items: []map[string]*dynamodb.AttributeValue{
+			{
+				"id": &dynamodb.AttributeValue{
+					S: aws.String("test.traefik.io_frontend"),
+				},
+				"name": &dynamodb.AttributeValue{
+					S: aws.String("test.traefik.io"),
+				},
+				"frontend": attributeFrontend,
+			},
+		},
+	}, true)
+	return nil
+}
+
+func TestLoadDynamoConfig(t *testing.T) {
+	dbiface := &dynamoClient{
+		db: &mockDynamoDBCLient{},
+	}
+	provider := DynamoDB{}
+	client := provider.createClient()
+	if client == nil {
```

Comment:
I don't think we need this check since we always return a non-nil value from `createClient`.